tests/src/core/testqgsrastermarker.cpp

```diff
@@ -68,35 +76,44 @@ class TestQgsRasterMarker : public QObject
     QgsSingleSymbolRenderer *mSymbolRenderer = nullptr;
     QString mTestDataDir;
     QString mReport;
+    QString mPercentageName;
 };
 
 
 void TestQgsRasterMarker::initTestCase()
 {
   mTestHasError = false;
-  // init QGIS's paths - true means that all path will be inited from prefix
+
+  // Init QGIS's paths - true means that all path will be inited from prefix
   QgsApplication::init();
   QgsApplication::initQgis();
   QgsApplication::showSettings();
 
-  //create some objects that will be used in all tests...
-  QString myDataDir( TEST_DATA_DIR ); //defined in CmakeLists.txt
+  // Create some objects that will be used in all tests...
+  QString myDataDir( TEST_DATA_DIR ); // defined in CmakeLists.txt
   mTestDataDir = myDataDir + '/';
 
-  //create a marker layer that will be used in all tests
+  // Create a marker layer that will be used in all tests
   QString pointFileName = mTestDataDir + "points.shp";
   QFileInfo pointFileInfo( pointFileName );
   mPointLayer = new QgsVectorLayer( pointFileInfo.filePath(),
                                     pointFileInfo.completeBaseName(), QStringLiteral( "ogr" ) );
 
   // Register the layer with the registry
-  QgsProject::instance()->addMapLayers(
-    QList<QgsMapLayer *>() << mPointLayer );
+  QgsProject::instance()->addMapLayers( QList<QgsMapLayer *>() << mPointLayer );
 
-  //setup the raster marker symbol
+  // Setup the raster marker symbol
   mRasterMarker = new QgsRasterMarkerSymbolLayer();
+  mRasterMarker->setPath( mTestDataDir + QStringLiteral( "sample_image.png" ) );
+  mRasterMarker->setSizeUnit( QgsUnitTypes::RenderPixels );
+  mRasterMarker->setSize( 30.0 );
+  mRasterMarker->setHorizontalAnchorPoint( QgsMarkerSymbolLayer::HCenter );
+  mRasterMarker->setVerticalAnchorPoint( QgsMarkerSymbolLayer::VCenter );
+  mRasterMarker->setOffsetUnit( QgsUnitTypes::RenderPixels );
```

Comment:
Same here -- the original version was more robust to failing tests

Reply:
Done.